src/atlannot/evaluation.py

```diff
@@ -0,0 +1,221 @@
+    """Compute the entropy of densities of Nissl for a given label value in the atlas.
+
+    Parameters
+    ----------
+    nissl: np.ndarray
+        Nissl volume.
+    atlas: np.ndarray
+        Annotation atlas.
+    label_value: int
+        Label Value to consider.
+    bins: int
+        Number of bins used to compute the histogram of the densities.
+
+    Returns
+    -------
+    region_entropy: float
+        Entropy of the densities of Nissl at a given region value.
+    """
+    prob_region, _ = np.histogram(nissl[atlas == label_value], bins=bins)
+    region_entropy = entropy(prob_region)
+    return region_entropy
+
+
+def compute_entropies(nissl, atlas):
+    """Compute entropies of Nissl densities.
+
+    Parameters
+    ----------
+    nissl: np.ndarray
+        Nissl volume.
+    atlas: np.ndarray
+        Annotation atlas.
+
+    Returns
+    -------
+    brain_entropy: float
+        Entropy of the densities of Nissl for the entire brain.
+
+    conditional_entropy: float
+        Conditional entropy of the densities of Nissl depending on the brain regions.
+    """
+    # 1: For the entire brain
+    prob_entire_brain, _ = np.histogram(nissl[atlas != 0], bins=100)
+    brain_entropy = entropy(prob_entire_brain)
+
+    # 2. Computation of the conditional entropy
+    n_pixels = (atlas != 0).sum()
+    label_values, count_values = np.unique(atlas, return_counts=True)
+    all_region_entropy = []
+    for label, count in zip(label_values, count_values):
+        all_region_entropy.append(compute_region_entropy(nissl, atlas, label) * count)
+
+    conditional_entropy = np.sum(all_region_entropy) / n_pixels
+    return brain_entropy, conditional_entropy
+
+
+def evaluate(
+    atlas,
+    nissl,
+    reference,
+    region_meta,
+    regions_to_evaluate=REGIONS_TO_EVALUATE,
+):
+    """Evaluate the atlas.
+
+    Parameters
+    ----------
+    atlas: np.ndarray
+        Atlas to evaluate.
+    nissl: np.ndarray
+        Corresponding Nissl Volume.
+    reference: np.ndarray
+        Reference atlas.
+    region_meta: atlannot.atlas.RegionMeta
+        Region Meta containing all the information concerning the labels.
+    regions_to_evaluate: dict[str, list[int]]
+        Regions to evaluate.
+
+    Returns
+    -------
+    results: dict[str, Any]
+        Dictionary containing the results of the evaluation.
+    """
+    results = {}
+    for name, region_ids in regions_to_evaluate.items():
+        desc = list(region_meta.descendants(region_ids))
+
+        # Put some metadata about the region
+        results[name] = {
+            "region_ids": region_ids,
+            "level": [region_meta.level[id_] for id_ in region_ids],
+            "descendants": desc,
+        }
+
+        # Jaggedness
+        values_from = np.unique(atlas)
+        values_to = np.array([1 if value in desc else 0 for value in values_from])
+        mask = atlas_remap(atlas, values_from, values_to)
```

Comment:
Is this maybe equivalent to `mask = np.isin(atlas, desc)`?

Reply:
Done in fd14718.

Reply:
Thanks! And do we still need `values_from` and `values_to` then?

Reply:
Sorry, my bad! Here are the new changes: 3f9aa78